php/query_utils.php

```diff
@@ -462,6 +462,139 @@ function get_associated_survey_instances(
+    if ($user["is_admin"]) {
+        $table_name[] = "dept_survey_choices";
+    }
+    if ($user["is_instructor"]) {
+        $table_name[] = "course_survey_choices";
+    }
+    if ($user["is_ta"]) {
+        $table_name[] = "ta_survey_choices";
+    }
+    $roles = [
+        "dept_survey_choices" => "dept_survey_choice_id",
+        "course_survey_choices" => "course_survey_choice_id",
+        "ta_survey_choices" => "ta_survey_choice_id"
+    ];
+
+    // unpack the query to be a regular list of ids
+    $ret = [];
+    foreach ($table_name as $t_name) {
+        // get all the choice ids from each table that corresponds to the user role
+        // Each survey has a set of choices from the department, course and individual TAs.
+        // Here we select a set of surveys that belong to a particular user,
+        // thus we use their role to get the id from the choice table that belong to their level.
+        //
+        //e.g. For admin with user_id turner32, we look into the dept_survey_choices table to their choice_id
+        //+----+------------+------------------+----------+
+        //| id | choices_id | department_name  | user_id  |
+        //+----+------------+------------------+----------+
+        //|  1 |          4 | Mathematics      | turner32
+        //
+        // in this example we have id 1
+        //
+        // Then in the surveys table, get the surveys with dept_survey_choice_id = 1
+        //
+        // survey_id | dept_survey_choice_id | course_survey_choice_id | ta_survey_choice_id
+        //         1 |                  NULL |                    NULL |                NULL
+        //         2 |                     1 |                    NULL |                NULL
+        //         3 |                  NULL |                    NULL |                   1
+        //         4 |                  NULL |                       1 |                   3
+        //         5 |                     2 |                       2 |                NULL
+        //         6 |                     1 |                    NULL |                NULL
+        //
+        // in the table above, surveys with survey_id 2,6 are returned for admin with user_id turner32
+        $col_name = $roles[$t_name];
+        $sql =
+            "SELECT survey_id FROM $t_name AS t " .
+            "JOIN surveys AS s " .
+            "ON s.$col_name = t.id " .
+            "WHERE t.user_id = :user_id AND s.term LIKE :term";
+        $bound = [
+            "user_id" => $user_id,
+            "term" => $term
+        ];
+        $res = do_select_query($sql, $bound, $conn);
+        foreach ($res->result as $x) {
+            $ret[] = $x["survey_id"];
+        }
+
+        $cond = "";
+        //if instructor we check if ta choice is also null
+        if ($t_name == "course_survey_choices") {
+            $cond = "AND ta_survey_choice_id IS NULL";
+
+            //if admin we check both instructor and ta choice are also null
```

Comment:
This part was not explained in the big comment.

Reply:
This is suppose to be with the big comment block below, I'll move it down and clarify it more.